### DISCUSSION TITLE
An approach to consider multiple slides per patient with weak labels.

### DISCUSSION THREAD

**Discussion by lysukhin** [jamesdolezal/slideflow#311] [Ideas]
Hello, @jamesdolezal and contributors!
First of all, let me thank you for this wonderful project that you develop and support. This work is really huge and important.

I have a question on possible plans to enable the following:
Usually, there is more than 1 histological slide per patient, but there's only 1 weak high-level label (e.g. malignant/benign) per patient. So the challenge is that not all of the patient's slides do actually provide meaningful information and features for this single label, like only 2 of 5 slides do really contain malignancy signals while the other 3 are actually 'clean'. Nevertheless, without additional annotations of each slide, this brings us to the same problem of noisy data as classical per-tile classification with weak labels.

So I wonder if there are any plans on how to deal with it, maybe in some shape of adding one more aggregating 'MIL'-level on top of features for all patients slides? So the model can choose among all feature bags, not just a set of them from a single slide.

Thanks for your attention and work, and looking forward to hear from you.

**Comment by jamesdolezal** [comment]
Thanks for initiating this discussion! I agree this is an important point we should consider. It should be fairly straightforward to provide users with the option to aggregate feature bags across slides for each patient when training MIL models. We'll scope out how quickly this can be implemented, but I'm assuming it shouldn't take more than a week or so to pull something together. I'll update this discussion when we have a dedicated branch with the feature.

**Reply by lysukhin** [reply]
Great, thanks for the feedback! I'd be happy to participate in case you have an understanding about the specific implementation of this functionality and need help.
